modules/apps/frontend-js/frontend-js-loader-support-api/src/main/java/com/liferay/frontend/js/loader/support/JSLoaderSupport.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.js.loader.support;
+
+import java.io.Writer;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+public interface JSLoaderSupport {
```

Comment:
This is the new proposed interface. I wanted to make something to replace `ScriptData`, as much as we can, in the future. This is because `ScriptData` is in `portal-kernel` so every time we need to change anything we break the world.

If we gradually migrate to this OSGi service we may be able to remove it in the mid/long term.

However, I'm not yet convinced of this and of the structure of this API given that it is still very tiny to envision anything near to its final structure.

Reply:
Nevertheless it is an internal API, we don't want to make it public for now, so we can -relatively- break it in the future.

Reply:
> However, I'm not yet convinced of this and of the structure of this API given that it is still very tiny to envision anything near to its final structure.
>
> Nevertheless it is an internal API, we don't want to make it public for now, so we can -relatively- break it in the future.

The API doesn't make sense to me... why a `jsLoaderSupport` thingie would have a `getScriptBody` and a `writeScript` method... it's like there's something missing for this to click and take us to the right API 😂 

I'm afraid Brian would be just as confused as we are now to refuse merging it, but we might provide a strong signal that this is only temporary and see if that gets through until we have a clearer idea.

Reply:
The problem is that I need `getScriptBody` for the template renderer project but because I'm not migrating the whole `ScriptData` yet I can't use the `writeScript` method. I can do one of:

1) Remove `writeScript` and give up trying to replace `ScriptData` with this.
2) Push it further and modify the API and the implementation so that the template renderer project can call `writeScript` too

I'll investigate 2 and if it looks complex I may rollback to 1 and tackle this again in a following PR.

Reply:
I wasn't against any method in particular, just that the combination of class and methods didn't make sense and didn't paint a clear picture of what this does

Reply:
I have succeeded in refactoring the API. I've had to copy-paste a bit of code from ScriptData to the React renderer but it will be removed once we get rid of AMD modules.